sql/2024/04/image-formats.sql

```diff
@@ -0,0 +1,146 @@
+                                                                                             'image/png',
+                                                                                             'image/jpeg',
+                                                                                             'image/avif',
+                                                                                             'image/gif',
+                                                                                             'image/bmp'));
+
+CREATE TEMPORARY FUNCTION
+  GET_LCP_ELEMENT_ATTRIBUTE(custom_metrics STRING,
+                            attribute STRING)
+  RETURNS STRING AS ( (
+  SELECT
+    JSON_EXTRACT_SCALAR(attr, "$.value") AS v
+  FROM
+    UNNEST(JSON_EXTRACT_ARRAY(JSON_EXTRACT(custom_metrics, '$.performance.lcp_elem_stats.attributes'))) AS attr
+  WHERE
+      JSON_EXTRACT_SCALAR(attr, "$.name") = attribute
+  LIMIT
+    1 ) );
+
+CREATE TEMPORARY FUNCTION
+  GET_LCP_IMAGE_ATTRIBUTE_VALUE(custom_metrics STRING,
+                                attribute STRING)
+  RETURNS STRING AS ( (
+  SELECT
+    JSON_EXTRACT_SCALAR(attributes, CONCAT("$.", attribute)) AS v
+  FROM
+    UNNEST(JSON_EXTRACT_ARRAY(custom_metrics, '$.Images')) AS attributes
+  WHERE
+    JSON_EXTRACT_SCALAR(attributes, CONCAT("$.", attribute)) IS NOT NULL
+    AND JSON_EXTRACT_SCALAR(attributes, CONCAT("$.", 'url')) = GET_LCP_ELEMENT_ATTRIBUTE(custom_metrics,
+                                                                                         'url')
+  LIMIT
+    1 ) );
+
+WITH
+  pagesWithLcpImages AS (
+    SELECT
+      date,
+      client,
+      page,
+      GET_LCP_ELEMENT_ATTRIBUTE(custom_metrics,
+                                'url') AS url,
+      GET_LCP_IMAGE_ATTRIBUTE_VALUE(custom_metrics,
+                                    'naturalWidth') AS image_width,
+      GET_LCP_IMAGE_ATTRIBUTE_VALUE(custom_metrics,
+                                    'naturalHeight') AS image_height
+    FROM
+      `httparchive.all.pages`
+    WHERE
+      IS_CMS(technologies,
+             'WordPress',
+             '')
+      AND LOWER(JSON_EXTRACT_SCALAR(custom_metrics, '$.performance.lcp_elem_stats.nodeName')) = 'img'
+      AND GET_LCP_IMAGE_ATTRIBUTE_VALUE(custom_metrics,
+                                        'naturalWidth') IS NOT NULL
+      AND GET_LCP_IMAGE_ATTRIBUTE_VALUE(custom_metrics,
+                                        'naturalHeight') IS NOT NULL
+      AND date = '2024-02-01' ),
+
+  imageRequests AS (
+    SELECT
+      date,
+      client,
+      page,
+      url,
+      LOWER(CAST(JSON_EXTRACT_SCALAR(summary, "$.mimeType") AS STRING)) AS mime_type,
+      CAST( JSON_EXTRACT_SCALAR(summary, "$.respSize") AS NUMERIC) AS resp_size,
+    FROM
+      `httparchive.all.requests`
+    WHERE
+      IS_IMAGE(summary)
+      AND date = '2024-02-01' )
+
+SELECT
+  date,
+  client,
+  mime_type,
```

Comment:
Per my comment in https://github.com/GoogleChromeLabs/wpp-research/pull/97#issuecomment-2052388273, maybe include here:
```suggestion
  mime_type,
  COUNT(url) AS num_lcp_images,
```

Reply:
@swissspidy Hmm, I tried this but I get tiny numbers, less than `10` everywhere. So that shows there's definitely a problem elsewhere. No way there are just less than 10 LCP images in this dataset :)

I'm not sure where that could come from. Potentially using `LIMIT 1` somewhere where it shouldn't be? Though at first glance I don't see where it would be a problem. We need to dig more.